internal/tests/edgegw/nat_rule_resource_test.go

```diff
@@ -0,0 +1,195 @@
+		resource.TestCheckResourceAttrSet(resourceName, "id"),
+		resource.TestCheckResourceAttrSet(resourceName, "edge_gateway_id"),
+		resource.TestCheckResourceAttrSet(resourceName, "external_address"),
+		resource.TestCheckResourceAttr(resourceName, "name", "example-dnat"),
+		resource.TestCheckResourceAttr(resourceName, "description", "description DNAT example"),
+		resource.TestCheckResourceAttr(resourceName, "internal_address", "11.11.11.4"),
+		resource.TestCheckResourceAttr(resourceName, "rule_type", "DNAT"),
+		resource.TestCheckResourceAttr(resourceName, "dnat_external_port", "8080"),
+		resource.TestCheckResourceAttr(resourceName, "priority", "0"),
+	)
+}
+
+func natRuleReflexiveTestCheck(resourceName string) resource.TestCheckFunc {
+	return resource.ComposeAggregateTestCheckFunc(
+		resource.TestCheckResourceAttrSet(resourceName, "id"),
+		resource.TestCheckResourceAttrSet(resourceName, "edge_gateway_id"),
+		resource.TestCheckResourceAttrSet(resourceName, "external_address"),
+		resource.TestCheckResourceAttr(resourceName, "name", "example-reflexive"),
+		resource.TestCheckResourceAttr(resourceName, "description", "description REFLEXIVE example"),
+		resource.TestCheckResourceAttr(resourceName, "internal_address", "192.168.0.1"),
+		resource.TestCheckResourceAttr(resourceName, "rule_type", "REFLEXIVE"),
+		resource.TestCheckResourceAttr(resourceName, "priority", "0"),
+	)
+}
+
+func TestAccNATRuleResource(t *testing.T) {
+	resourceName := "cloudavenue_edgegateway_nat_rule.example"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { tests.TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: tests.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Read testing
+			{
+				// Apply test Snat
+				Config: tests.ConcatTests(testAccNATRuleResourceConfigSnat, testAccNATRuleResourceConfigDataSource),
+				Check:  natRuleSnatTestCheck(resourceName),
+			},
+			// Update testing Snat
+			{
+				// Update test
+				Config: tests.ConcatTests(testAccNATRuleResourceConfigUpdateSnat, testAccNATRuleResourceConfigDataSource),
+				Check:  natRuleSnatUpdateTestCheck(resourceName),
+			},
+			{
+				// Apply test Dnat
+				Config: tests.ConcatTests(testAccNATRuleResourceConfigDnat, testAccNATRuleResourceConfigDataSource),
+				Check:  natRuleDnatTestCheck(resourceName),
+			},
+			{
+				// Delete test
+				Destroy: true,
+				Config:  tests.ConcatTests(testAccNATRuleResourceConfigSnat, testAccNATRuleResourceConfigDataSource),
+			},
+			{
+				// Apply test Reflexive
+				Config: tests.ConcatTests(testAccNATRuleResourceConfigReflexive, testAccNATRuleResourceConfigDataSource),
+				Check:  natRuleReflexiveTestCheck(resourceName),
+			},
+			// Import State testing
+			{
+				// Import test
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccNATRuleResourceImportStateIDFunc(resourceName),
+			},
```

Comment:
Add one test for testing import edgeGatewayID and NatRuleID

Reply:
Add test with invert attributes asked